api/src/main/java/jakarta/persistence/PersistenceConfiguration.java

```diff
@@ -76,27 +198,35 @@ public class PersistenceConfiguration {
     private ValidationMode validationMode = ValidationMode.AUTO;
     private PersistenceTransactionType transactionType = PersistenceTransactionType.RESOURCE_LOCAL;
 
-    private List<Class<?>> managedClasses = new ArrayList<>();
-    private List<String> mappingFileNames = new ArrayList<>();
-    private Map<String,Object> properties = new HashMap<>();
+    private final List<Class<?>> managedClasses = new ArrayList<>();
+    private final List<String> mappingFileNames = new ArrayList<>();
```

Comment:
It might be better to have this defined as `List<java.nio.file.Path>` instead of strings which in the end the provider needs to convert to something usable for getting a stream out of it. With `Path` we'd only need to resolve it against "some" root, if it is not absolute and then use sth like:

```
try (InputStream is = Files.newInputStream(mappingFile) {
    parse(is);
}
```

to read its content. This leaving necessary _String_ validation on the caller - and that is IMO not a bad thing

Reply:
Ummm OK so I guess the question is: are these files loaded using `getResource()`, or using a `File`?

Reply:
So I'm checking our APIs. We have:

```java
	/**
	 * Read mappings as an application resourceName (i.e. classpath lookup).
	 *
	 * @param name The resource name
	 *
	 * @return this (for method chaining purposes)
	 */
	public MetadataSources addResource(String name) {
		...
		return this;
	}

	/**
	 * Read mappings from a particular XML file.
	 * <p>
	 * The given path is resolved using {@link File#File(String)}.
	 *
	 * @param path The path to a file
	 *
	 * @return this (for method chaining purposes)
	 *
	 * @see #addFile(File)
	 */
	public MetadataSources addFile(String path) {
		addFile( new File( path ) );
		return this;
	}

	/**
	 * Read mappings from a particular XML file.
	 *
	 * @param file The reference to the XML file
	 *
	 * @return this (for method chaining purposes)
	 */
	public MetadataSources addFile(File file) {
		...
		return this;
	}
```

So we let you add a `File`, a `String` interpreted as a resource, or a `String` interpreted as a file path.

Reply:
> Ummm OK so I guess the question is: are these files loaded using `getResource()`, or using a `File`?

good point. The spec as well as the schema explicitly state that mapping-files are loaded _as resources_ which makes it obvious that _String_  interpreted as a resource name is the only variant to be  available (unless we allow alternatives also directly in the schema/spec), my bad. The only restriction we currently have in place is about the exact location of _orm.xml_

Reply:
let's not add File/Path variants and keep that on the providers for now